### ISSUE TITLE
auto_mixed_precision slow due to "Sum" being on blacklist.

### ISSUE THREAD

**Issue by reedwm** [tensorflow/tensorflow#28298]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): 30d87b5299a393cd0608ec1181af9f4529065749
- Python version: 2.7
- Bazel version (if compiling from source): 0.24.1
- GCC/Compiler version (if compiling from source): 5.4
- CUDA/cuDNN version: 10/7
- GPU model and memory: 8 V100s

**Describe the current behavior**

[Sum is on the blacklist](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h#L168) for the auto_mixed_precision grappler pass. Unfortunately, Sum is used in [the gradient of tf.add](https://github.com/tensorflow/tensorflow/blob/63db602e38c0c82bed9dfcf96f7df2b2041ffb20/tensorflow/python/ops/math_grad.py#L1008), if non-static shapes are used. This greatly slows down [resnet50](https://github.com/tensorflow/models/tree/master/official/resnet), from 709 to 551 images/sec. The Sums are done in fp32, and relus and batchnorm gradients are fp32 as well.

Using static shapes fixes this issue, but many users use dynamic shapes with tf.add.

Sum was added to the blacklist in 180542ba39fdb4a61cbe7a92d7ef512526383dea. The justification is "The dice loss function used in U-net uses Sums that overflow in fp16"

**Describe the expected behavior**

Sum should not be on the blacklist, and resnet50 should get 709 images/sec. Maybe we can find a way to do the dice loss function Sum in fp32, but other Sums in fp16?

**Code to reproduce the issue**
N/A

**Other info / logs**
N/A

/CC @azaks2 @tfboyd @nluehr @benbarsdell @MattConley

**Comment by benbarsdell** [comment]
https://github.com/tensorflow/tensorflow/pull/28057 is supposed to help with this; can you see if it improves performance in this case?

**Comment by reedwm** [comment]
I'll wait until the PR is merged, then try.

**Comment by reedwm** [comment]
This issue is fixed by #28057
